cmd2/cmd2.py

```diff
@@ -3332,18 +3332,17 @@ def do_history(self, args: argparse.Namespace) -> None:
             for hi in history:
                 self.poutput(hi.pr(script=args.script, expanded=args.expanded, verbose=args.verbose))
 
-    def _generate_transcript(self, history: List[HistoryItem], transcript_file: str) -> None:
+    def _generate_transcript(self, history: List[Union[HistoryItem, str]], transcript_file: str) -> None:
         """Generate a transcript file from a given history of commands."""
-        # Save the current echo state, and turn it off. We inject commands into the
-        # output using a different mechanism
         import io
 
+        # Disable echo and redirection while we manually redirect stdout to a StringIO buffer
```

Comment:
I don't think we should be setting `self.allow_redirection` here. That will change the meaning of redirection characters in the commands being loaded from the script if `allow_redirection` is typically False. Also we already overwriting `self.stdout` so redirection is being done anyway.

Reply:
I was debating about that.  I was doing this defensively to prevent an unhandled exception if we didn't have it and someone did something like:
```sh
(Cmd) load script.txt -r transcript.txt > out.txt
```

Can you think of a better way to defend against that?

Reply:
If redirection was off, then I think that command would result in an `argparse` error.

Reply:
Correction, if redirection if off, then '> out.txt' just gets ignored by the `StatementParser`.

Reply:
So what are your thoughts on how to best deal with this?

Reply:
I'm thinking of just putting a guard clause in the ``history -t`` and ``load -r`` cases where if ``self.redirecting`` is true, we bail with an error message.